libglusterfs/src/glusterfs/logging.h

```diff
@@ -352,11 +352,15 @@ _gf_smsg(const char *domain, const char *file, const char *function,
     } while (0)
 
 /* Shortcut macros for different log levels. */
-#define GF_LOC_C(_name, _data) GF_LOG(_name, GF_LOG_CRITICAL, _data)
+#define GF_LOG_C(_name, _data) GF_LOG(_name, GF_LOG_CRITICAL, _data)
 #define GF_LOG_E(_name, _data) GF_LOG(_name, GF_LOG_ERROR, _data)
 #define GF_LOG_W(_name, _data) GF_LOG(_name, GF_LOG_WARNING, _data)
 #define GF_LOG_I(_name, _data) GF_LOG(_name, GF_LOG_INFO, _data)
 
+/* 'global_ctx' is defined in glusterfs/globals.h. However if we include it
+ * here it will create a include loop. For now, anyone that uses these macros
+ * needs to include glusterfs/globals.h before using them. */
+
 #define GF_LOG_D(_name, _msg, _num, _fields...)                                \
```

Comment:
I wonder if we should move to looking at global_ctx in skip_logging() as well.

Reply:
Probably we should rework the includes as a whole because we already have several circular dependencies.

One idea I've been thinking is to create a 3-level include structure:

1. First level just declares all types (only declarations, not definitions). It can be a single file or it can be split into multiple files.
2. Second level defines types. In general there would be one file per type, but closely coupled types can be defined in the same file.
3. Third level declares functions.

To avoid circular dependencies, an include in one level can only include other includes from lower levels (i.e. functions can include type definitions and type declarations, and type definitions can only include type declarations).